src/App/Listener/HandleTargetedSurveyResponse.php

```diff
@@ -0,0 +1,143 @@
+<?php
+namespace Ushahidi\App\Listener;
+
+use Ushahidi\Core\Entity\Message;
+use Ushahidi\Core\Entity\MessageRepository;
+use Ushahidi\Core\Entity\TargetedSurveyState;
+use Ushahidi\Core\Entity\TargetedSurveyStateRepository;
+use Ushahidi\Core\Entity\FormAttributeRepository;
+
+class HandleTargetedSurveyResponse
+{
+
+    protected $messageRepo;
+    protected $contactRepo;
+    protected $formAttrRepo;
+
+    /**
+     * Create the event listener.
+     *
+     * @return void
+     */
+    public function __construct(
+        MessageRepository $messageRepo,
+        TargetedSurveyStateRepository $targetedSurveyStateRepo,
+        FormAttributeRepository $formAttrRepo
+    ) {
+        $this->targetedSurveyStateRepo = $targetedSurveyStateRepo;
+        $this->messageRepo = $messageRepo;
+        $this->formAttrRepo = $formAttrRepo;
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @return void
+     */
+    public function handle($incomingMessageId, $incomingMessage, $inbound_form_id, $inbound_fields)
+    {
+        // \Log::info('Running HandleTargetedSurveyResponse', func_get_args());
+
+        // check if contact is part of an open targeted_survey
+        if (!$this->targetedSurveyStateRepo
+            ->isContactInActiveTargetedSurveyAndReceivedMessage($incomingMessage->contact_id)
```

Comment:
This was previously done in a 3 stage check where we:
1. checked if the contact was in a survey
2. Loaded the survey state
3. checked if there was an outbound message.

I moved that all into a single SQL query in the repo.